### PR TITLE
fix(transaction-pool): tx broadcast fixes

### PR DESCRIPTION
**Pull request by oXtxNt9U**
## Checklist

- [ ] Documentation _(if necessary)_
- [ ] Tests _(if necessary)_
- [ ] Ready to be merged

### CODE REVIEW

File: packages/crypto-block/source/verifier.ts [hunk 1]
```diff
@@ -92,7 +92,7 @@ export class Verifier implements Contracts.Crypto.IBlockVerifier {
 
 			let totalAmount: BigNumber = BigNumber.ZERO;
 			let totalFee: BigNumber = BigNumber.ZERO;
-			let totalPayloadLength = 0;
+			let totalPayloadLength = 4 * block.transactions.length;
```
Comment (sebastijankuzner) [inline packages/crypto-block/source/verifier.ts:95]: Why is this changed?

File: packages/state/source/transaction-validator.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ export class TransactionValidator implements Contracts.State.TransactionValidato
 	private readonly handlerRegistry!: Contracts.Transactions.ITransactionHandlerRegistry;
 
 	@inject(Identifiers.WalletRepository)
-	@tagged("state", "blockchain")
+	@tagged("state", "copy-on-write")
```
Comment (sebastijankuzner) [inline packages/state/source/transaction-validator.ts:11]: Double check if this tag should be **clone**.

File: packages/crypto-block/source/verifier.ts (hunk 1 above)
Comment (oXtxNt9U) [reply packages/crypto-block/source/verifier.ts:95]: The payload length was calculated wrong here and for empty blocks this change makes no difference, hence it wasn't spotted earlier.

File: packages/state/source/transaction-validator.ts (hunk 2 above)
Comment (oXtxNt9U) [reply packages/state/source/transaction-validator.ts:11]: Yes, or rather it depends.

At the moment the transaction validator is only used by the pool, which is only supposed to apply transactions on the copy-on-write wallet repo. Ideally, this would be tagged dynamically, but I couldn't make it work.

File: packages/crypto-block/source/verifier.ts (hunk 1 above)
Comment (oXtxNt9U) [reply packages/crypto-block/source/verifier.ts:98]: added a comment for clarification

```typescript
result.writeUint32(serialized.length);
result.writeBytes(serialized);
```

File: packages/state/source/transaction-validator.ts (hunk 2 above)
Comment (oXtxNt9U) [reply packages/state/source/transaction-validator.ts]: Nevermind my previous comment.  Should be `clone`
